blocks/monitoring/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""This part of the framework helps you monitor your model during training.
+
+:class:`MonitorChannels` describe quantities that should be monitored. They
+can either be Theano variables (e.g. the objective function, the weight
+norms, etc.) or they can be callable functions (e.g. if you want to use
+NumPy to calculate eigenvalues, or sample text from your language model).
+
+.. todo::
+
+   Monitored Theano variables might depend on inputs beyond those given to
+   the model. There should be a way of providing these to the monitoring
+   channels, collecting them, and compiling them as part of the Theano
+   function that performs monitoring.
+
+"""
+import logging
+from abc import ABCMeta, abstractmethod
+
+from blocks.utils import shared_expression, update_instance
+
+logger = logging.getLogger(__name__)
+
+
+class MonitorChannel(object):
```

Comment:
Okay, I think we actually agree, I was just thoroughly confused by your use of the `ModelProperty` as an `AggregationScheme` and on how you expected people to use it, but if you just use it internally to simplify code and add it silently it makes more sense perhaps.

I'll change the names back and maybe write some documentation. Also, if your idea is just to add some information on the **how** to aggregate, we don't need to name aggregators, instead we can just name the resulting expressions using the `add_monitor` method.